modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java

```diff
@@ -280,6 +286,8 @@ public void processOpts() {
         } else if (RESTTEMPLATE.equals(getLibrary())) {
             additionalProperties.put("jackson", "true");
             supportingFiles.add(new SupportingFile("auth/Authentication.mustache", authFolder, "Authentication.java"));
+        } else if (WEBCLIENT.equals(getLibrary())) {
+            additionalProperties.put("jackson", "true");
```

Comment:
here I would set other values like `java8Mode`, `dateLibrary`...

Reply:
At first I added this check at line 290, but it was not working, because dateLibrary is used in AbstractJavaCodegen.processOpts(), so setting it on line 290 doesn't work.

As you stated, setting it on line 156 doesn't always work either (because it's set inside AbstractJavaCodegen.processOpts if parameter about date library is present).

I think, the only solution is to divide processOpts(). Currently it does some things:
* gets params and sets fiels (e.g dateLibrary)
* does some actions according to fields
For example, in the end of processOpts:
```java
        if ("threetenbp".equals(dateLibrary)) {
            additionalProperties.put("threetenbp", "true");
            additionalProperties.put("jsr310", "true");
            typeMapping.put("date", "LocalDate");
            typeMapping.put("DateTime", "OffsetDateTime");
            importMapping.put("LocalDate", "org.threeten.bp.LocalDate");
            importMapping.put("OffsetDateTime", "org.threeten.bp.OffsetDateTime");
        } else if ("joda".equals(dateLibrary)) {
```

Possibly, this method (processOpts) should do only one thing (should be divided)
and the method for checking/updating fields should be added.

Actually, it's not a big deal now. I can add back threetenbp support (but it's useless cause java8 support only), then there is no reason to do all these things now.

What do you think?

Reply:
Thank you a lot for the analysis and the explanations.

I think that we can do the refactoring you have proposed (splitting `processOpts()`) in a separated PR.